File: frontend/services/courseService.js
import api from "./api";

export const getCourses = (params) => {
  return api.get("/api/courses", {
    params: {
      page: params.page,
      size: params.size,
      search: params.search,
      sortBy: params.sortBy,
      sortDir: params.sortDir,
    },
  });
};

export const createCourse = (data) => {
  return api.post("/api/courses", data);
};

export const updateCourse = (id, data) => {
  return api.put(`/api/courses/${id}`, data);
};

export const deleteCourse = (id) => {
  return api.delete(`/api/courses/${id}`);
};

File: Backend/src/main/java/com/assignment/courseManagementSystem/services/contract/CourseService.java
package com.assignment.courseManagementSystem.services.contract;

import com.assignment.courseManagementSystem.dto.CourseDto;
import org.springframework.data.domain.Page;

public interface CourseService {

    CourseDto createCourse(CourseDto dto);

    CourseDto getCourseById(Long id);

    Page<CourseDto> getCourses(
            String search,
            int page,
            int size,
            String sortBy,
            String sortDir
    );

    CourseDto updateCourse(Long id, CourseDto dto);

    void deleteCourse(Long id);
}
